source/kraken/apply_disruption.cpp

```diff
@@ -310,16 +324,16 @@ void delete_disruption(const std::string& disruption_id,
                        const nt::MetaData& meta) {
     auto log = log4cplus::Logger::getInstance("log");
     LOG4CPLUS_DEBUG(log, "Deleting disruption: " << disruption_id);
-    nt::disruption::DisruptionHolder& holder = pt_data.disruption_holder;
 
+    Finally finally{[&](){
+        pt_data.disruption_holder.clean_weak_impacts();
+    }};
```

Comment:
Please add a comment explaining that Finally is needed because we want to call `clean_weak_impacts` after `disruption` deletion.